release-process.rst

```diff
@@ -28,55 +34,178 @@ Roughly:
+   with expectations or similar functionality. The potential impact of such a
+   fix SHOULD be evaluated, and based on that, the change MAY be treated as a BC
+   break if it is likely to affect real-world code in significant or disruptive
+   ways.
+
+On Breaking Compatibility
+-------------------------
+
+Breaking BC, APIs, or ABIs (internal) SHOULD NOT be done lightly or for the sake
+of doing it. RFCs explaining the reasoning behind a breakage MUST include:
+
+-  Clear justification for the change
+-  A summary of its consequences
+-  Migration guidance
+-  Test cases and patches
+
+If a high severity security fix requires breaking the internal ABI or API, a
+proper migration path MUST be provided, and the impact MUST be minimized as much
+as possible. This MUST also be accompanied by additional communication during
+the release.
+
+SAPI removal SHOULD generally occur only in a major version. However, in
+exceptional cases where a SAPI presents significant maintenance or security
+concerns, an exception MAY be made. In all cases, the removal MUST go through
+the RFC process.
+
+All new user-facing features MUST be mentioned in the `UPGRADING
+<https://github.com/php/php-src/blob/master/UPGRADING>`_ document.
+
+All API and ABI breaks MUST be mentioned in the `UPGRADING.INTERNALS
+<https://github.com/php/php-src/blob/master/UPGRADING.INTERNALS>`_ document.
+
+Further reading:
+
+-  http://en.wikipedia.org/wiki/Application_programming_interface
+-  http://en.wikipedia.org/wiki/Application_binary_interface
 
 Major Version Number
 ====================
 
 -  x.y.z to x+1.0.0
 
-   -  Bug fixes
-   -  New features
-   -  Extensions support can be ended (moved to PECL)
-   -  Backward compatibility can be broken
-   -  API compatibility can be broken (internals and userland)
-   -  ABI can be broken (internals)
+   -  It SHOULD include bugfixes and new features.
+   -  Extensions support MAY be ended (moved to PECL).
+   -  SAPI support MAY be ended (removed from the php-src repository).
+   -  ABI backward compatibility MAY be broken.
+   -  Internal API backward compatibility MAY be broken.
+   -  Userland API backward compatibility MAY be broken.
+   -  Significant userland API backward compatibility breaks SHOULD be preceded
+      by the deprecation phase in the previous major version.
 
 Minor Version Number
 ====================
 
--  x.y.z to x.y+1.z
+-  x.y.z to x.y+1.0
 
-   -  Bugfixes
-   -  New features
-   -  Extensions support can be ended (moved to PECL)
-   -  Backward compatibility must be kept
-   -  API compatibility must be kept (userland)
-   -  ABI and API can be broken (internals)
-   -  Source compatibility should be kept if possible, while breakages are
-      allowed
+   -  It SHOULD include bugfixes and new features.
+
+   -  Extensions support MAY be ended (moved to PECL).
+
+   -  SAPI support is RECOMMENDED to be kept.
+
+   -  Internal API compatibility breaks are NOT RECOMMENDED.
```

Comment:
This does not follow current practice IMHO. We break internal API regularly in minor versions.

Reply:
Ok I changed it so it's:

```
   -  Internal API compatibility MAY be broken when necessary (for example,
      during refactoring), but SHOULD be preserved whenever possible, such as by
      adding an _ex variant of a function instead of changing the existing one.
```

Does this sound ok to you?

Reply:
This does sound better to me, thank you!